File: sandbox-runner/patterns/continue-as-new/go/starter.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"go.temporal.io/sdk/client"
)

func main() {
	c, err := client.Dial(client.Options{HostPort: "localhost:7233"})
	if err != nil {
		log.Fatalln("Unable to create client:", err)
	}
	defer c.Close()

	workflowID := fmt.Sprintf("%s-%d", WorkflowIDPrefix, time.Now().UnixMilli())
	we, err := c.ExecuteWorkflow(
		context.Background(),
		client.StartWorkflowOptions{
			ID:        workflowID,
			TaskQueue: TaskQueue,
		},
		DataProcessorWorkflow,
		"",
		0,
	)
	if err != nil {
		log.Fatalln("Unable to execute workflow:", err)
	}
	fmt.Printf("Started workflow: %s\n", we.GetID())

	var total int
	if err := we.Get(context.Background(), &total); err != nil {
		log.Fatalln("Workflow failed:", err)
	}
	fmt.Printf("Final result: processed %d records\n", total)
	fmt.Printf("Open the Temporal UI and search for '%s' to see the Continue-As-New chain.\n", workflowID)
}

File: sandbox-runner/patterns/continue-as-new/java/src/main/java/Starter.java
import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowOptions;
import io.temporal.serviceclient.WorkflowServiceStubs;

public class Starter {
    public static void main(String[] args) {
        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
        WorkflowClient client = WorkflowClient.newInstance(service);

        String workflowId = Shared.WORKFLOW_ID_PREFIX + "-" + System.currentTimeMillis();
        DataProcessorWorkflow workflow = client.newWorkflowStub(
                DataProcessorWorkflow.class,
                WorkflowOptions.newBuilder()
                        .setTaskQueue(Shared.TASK_QUEUE)
                        .setWorkflowId(workflowId)
                        .build());

        System.out.println("Started workflow: " + workflowId);
        int total = workflow.process("", 0);
        System.out.println("Final result: processed " + total + " records");
        System.out.println("Open the Temporal UI and search for '" + workflowId
                + "' to see the Continue-As-New chain.");
    }
}
